GETTING_STARTED.md

```diff
@@ -0,0 +1,302 @@
+- Include exception handling in sensitive sections such as callbacks.
+- Checkpoint/save prompts and received responses periodically.
+- Check examples in `examples/user` directory and modify them for your needs.
+
+```python
+def my_callback(result):
+    try:
+        ...
+    except:
+        ...
+
+outputs = []
+count = 0
+for result in model.generate_async(prompts, callback=my_callback):
+    if result is not None:
+        print(result.input_text, " --> ", result.generated_text)
+        # check if prompts[count] and result.input_text are the same
+        outputs.append((result.input_text, result.generated_text))
+        # periodically save outputs to disk or some location
+        ...
+    else:
+        # ... save failed prompts for retrying
+    count += 1
+```
+
+## <a name='Extensions'></a>Extensions
+
+GenAI currently supports a langchain extension and more extensions are in the pipeline. Please reach out to
+us if you want support for some framework as an extension or want to design an extension yourself.
+
+### <a name='LangChainExtension'></a>LangChain Extension
+
+Install the langchain extension as follows:
+```bash
+pip install "ibm-generative-ai[langchain]"
+```
+Currently the langchain extension allows IBM Generative AI models to be wrapped as Langchain LLMs and translation between genai PromptPatterns and LangChain PromptTemplates. Below are sample snippets
+```python
+import os
+from dotenv import load_dotenv
+import genai.extensions.langchain
+from genai.extensions.langchain import LangChainInterface
+from genai.schemas import GenerateParams
+from genai import Credentials, Model, PromptPattern
+
+load_dotenv()
+api_key = os.getenv("GENAI_KEY", None)
+api_endpoint = os.getenv("GENAI_API", None)
+creds = Credentials(api_key, api_endpoint)
+params = GenerateParams(decoding_method="greedy")
+
+# As LangChain Model
+langchain_model = LangChainInterface(model="google/flan-ul2", params=params, credentials=creds)
+print(langchain_model("Answer this question: What is life?"))
+
+# As GenAI Model
+genai_model = Model(model="google/flan-ul2", params=params, credentials=creds)
+print(genai_model.generate(["Answer this question: What is life?"])[0].generated_text)
+
+# GenAI prompt pattern to langchain PromptTemplate and vice versa
+seed_pattern = PromptPattern.from_str("Answer this question: {{question}}")
+template = seed_pattern.langchain.as_template()
+pattern = PromptPattern.langchain.from_template(template)
+
+print(langchain_model(template.format(question="What is life?")))
+print(genai_model.generate([pattern.sub("question", "What is life?")])[0].generated_text)
+```
+
+## <a name='Model Types'></a>Model Types
+
+Model types can be imported from the [ModelType class](src/genai/schemas/models.py). If you want to use a model that is not included in this class, you can pass it as a string as exemplified [here](src/genai/schemas/models.py).
```

Comment:
Since we are deprecating ModelType maybe let's say something like:
```suggestion
## <a name='[Deprecated] Model Types'></a>[Deprecated] Model Types

Model types can be imported from the [ModelType class](src/genai/schemas/models.py). If you want to use a model that is not included in this class, you can pass it as a string as exemplified [here](src/genai/schemas/models.py).

Models can be selected by passing their string id to the Model class as exemplified [here](src/genai/schemas/models.py).
```